src/epics/data.ts

```diff
@@ -970,22 +975,69 @@ export const removeMultivalueTag: Epic = (action$, store) => action$.ofType(type
         }
     }
 
-    return Observable.concat(
-        Observable.of($do.changeDataItem({
-            bcName: removedItemBc,
-            cursor: removedItemId,
-            dataItem: {
-                ...(action.payload.removedItem as any),
-                _associate: false,
+    if (parentDepth && widget.options?.hierarchyTraverse) {
+        const parentItem = (removeDataItem.level > 1) && allData?.find((item) => item.id === removeDataItem.parentId)
+        const removedItem = action.payload.dataItem.find(item => item.id === parentItem.id)
+        const currentLevelData = allData?.filter(
+            (item) => item.level === removeDataItem?.level && (item.level === 1 || (item.parentId === removeDataItem?.parentId))
+        )
+        const currentLevelDelta = delta.filter(
+            (item) => item.level === removeDataItem?.level && (item.level === 1 || (item.parentId === removeDataItem?.parentId))
+        )
+        const wasLastInDelta = !currentLevelDelta
+            .filter(item => item.id !== removeDataItem.id)
+            .filter(item => item.level === removeDataItem.level).length
+        // delta include only associated items and contain associated dataItems
+        const wasLastInData = delta?.length > 0 || currentLevelData
+            .filter(item => item.id !== removeDataItem.id)
+            .every(item => item._associate === false)
+
+        if (removedItem && wasLastInData && wasLastInDelta) {
+            result.push(Observable.of($do.removeMultivalueTag({
+                bcName: action.payload.bcName,
+                popupBcName: action.payload.popupBcName,
+                cursor: action.payload.cursor,
+                associateFieldKey: action.payload.associateFieldKey,
+                dataItem: action.payload.dataItem.filter(item => item.id !== parentItem.id),
+                removedItem: removedItem
+            })))
+        }
+    }
+
+    if (widget.options?.hierarchyGroupDeselection) {
+        // delta include only associated items and contain associated dataItems
+        const currentData = delta.length > 0 ? [] : allData.filter(item => item._associate === true)
+        const currentDataItems = [...currentData, ...delta]
+        const childDataItems = currentDataItems.filter(item => item.parentId === removeDataItem.id)
+        const deleteItems: DataItem[] = []
+        while(childDataItems.length > 0){
+            const tempItem = childDataItems.shift()
```

Comment:
This approach does not match `hierarchyTraverse` option: it'll fetch associated descendants of associated nodes, but fill not fetch associated descendants of unassociated nodes.  

I think without `hierarchyTraverse` option it should only fetch direct children; otherwise it should fetch all associated descdendants, without skipping intermidiate unassociated nodes